M5StackCommon/Core2ToughCommon.cs

```diff
@@ -0,0 +1,479 @@
+
+            // Sets the Vibrator voltage
+            _power.LDO3OutputVoltage = ElectricPotential.FromVolts(2.0);
+            // vibrator off
+            Vibrate = false;
+
+            // Sets the LCD backlight voltage to 2.8V
+            _power.DcDc3Voltage = ElectricPotential.FromVolts(2.8);
+
+            // GPIO1 set to PWM to control power LED
+            _power.Gpio1Behavior = Gpio12Behavior.PwmOutput;
+
+            // Switch on the power LED
+            PowerLed = true;
+
+            // battery = 360mAh
+            _power.ChargingCurrent = ChargingCurrent.Current360mA;
+
+#elif TOUGH
+            // PWM1 X
+            _power.Pwm1OutputFrequencySetting = 0;
+            // PWM1 Y1
+            _power.Pwm1DutyCycleSetting1 = 0xFF;
+            // PWM1 Y2
+            _power.Pwm1DutyCycleSetting2 = 0xFF;
+
+            // enable DCO and LDO outputs
+            _power.LdoDcPinsEnabled = LdoDcPinsEnabled.DcDc1 | LdoDcPinsEnabled.Ldo2 | LdoDcPinsEnabled.Ldo3;
+            
+            // Sets the LCD backlight voltage to 3V
+            _power.LDO3OutputVoltage = ElectricPotential.FromVolts(3.0);
+
+              // GPIO1 is reset for Touch controller
+            _power.Gpio1Behavior = Gpio12Behavior.MnosLeakOpenOutput;
+
+#endif
+
+            // Setup buttons
+            _gpio = new();
+
+            // Setup SPI1
+            Configuration.SetPinFunction(23, DeviceFunction.SPI1_MOSI);
+            Configuration.SetPinFunction(38, DeviceFunction.SPI1_MISO);
+            Configuration.SetPinFunction(18, DeviceFunction.SPI1_CLOCK);
+
+            // Setup the screen with SP2 and SD Card
+            Configuration.SetPinFunction(23, DeviceFunction.SPI2_MOSI);
+            Configuration.SetPinFunction(38, DeviceFunction.SPI2_MISO);
+            Configuration.SetPinFunction(18, DeviceFunction.SPI2_CLOCK);
+
+            // Second serial port
+            Configuration.SetPinFunction(13, DeviceFunction.COM2_RX);
+            Configuration.SetPinFunction(14, DeviceFunction.COM2_TX);
+
+            // Setup second I2C bus (port A) 
+            Configuration.SetPinFunction(33, DeviceFunction.I2C2_CLOCK);
+            Configuration.SetPinFunction(32, DeviceFunction.I2C2_DATA);
+            // The portA is the second I2C
+            _portANumber = 2;
+
+            // Setup the time if any
+            _rtc = new Pcf8563(I2cDevice.Create(new I2cConnectionSettings(1, Pcf8563.DefaultI2cAddress)));
+
+            DateTime dt;
+            var sysDtcore = DateTime.UtcNow;
+            try
+            {
+                dt = _rtc.DateTime;
+                if (sysDtcore < dt)
+                {
+                    Rtc.SetSystemTime(dt);
+                }
+            }
+            catch (Exception)
+            {
+
+                if (sysDtcore.Year < 2021)
+                {
+                    dt = new DateTime(2021, 11, 03, 12, 00, 00);
```

Comment:
maybe increase to today? :-)

Reply:
To be honest I didn't even looked at this line... Done!